tracer/src/Datadog.Trace.Coverage.collector/AssemblyProcessor.cs

```diff
@@ -722,13 +724,25 @@ public override AssemblyDefinition Resolve(AssemblyNameReference name)
                         _logger.Debug($"Looking for: {pathTest}");
                         if (File.Exists(pathTest))
                         {
-                            return AssemblyDefinition.ReadAssembly(pathTest);
+                            try
+                            {
+                                return AssemblyDefinition.ReadAssembly(pathTest);
+                            }
+                            catch (Exception innerAssemblyException)
+                            {
+                                _logger.Error(innerAssemblyException, $"Error reading the assembly: {pathTest}");
+                                throw;
+                            }
                         }
 
-                        _logger.Error(ex, $"Error in the Custom Resolver processing '{_assemblyFilePath}' for: {name.FullName}");
+                        _logger.Error(arEx, $"Error in the Custom Resolver processing '{_assemblyFilePath}' for: {name.FullName}");
```

Comment:
nit: don't use interpolation in log messages
```suggestion
                        _logger.Error(arEx, "Error in the Custom Resolver processing '{Path}' for: {AssemblyName}", _assemblyFilePath, name.FullName);
```

Reply:
The logger used here is not the same:

https://github.com/DataDog/dd-trace-dotnet/blob/e064fdb419694a846f2a178658d4726c7b806c79/tracer/src/Datadog.Trace.Coverage.collector/ICollectorLogger.cs